src/utilities/add-remove.test.js

```diff
@@ -0,0 +1,203 @@
+  };
+
+  const addFunction = add(event);
+
+  //   Assert --------------------------------------------------------------------------->
+  test('add task to the list', () => {
+    expect(addFunction.length === 1).toBeTruthy();
+  });
+
+  test('task contains a right index', () => {
+    expect(addFunction[addFunction.length - 1].index).toBe(0);
+  });
+
+  test('add a object', () => {
+    expect(addFunction).toEqual([
+      {
+        description: 'Hello',
+        status: false,
+        index: 0,
+      },
+    ]);
+  });
+
+  test('add task to DOM', () => {
+    expect(checkElementsInTaskDOM()[3]).toBeDefined();
+  });
+
+  test('test get localStorage', () => {
+    expect(localStorage.getDataLocalStorage()).toEqual([
+      {
+        description: 'Hello',
+        status: false,
+        index: 0,
+      },
+    ]);
+  });
+});
+
+describe('Test remove', () => {
+  // Arrange --------------------------------------------------------------------------->
+  document.body.innerHTML = '<div>'
+     + '<input type="text" name="task" id="task" class="input-task" value="Hello">'
+     + '<ul class="container-list">'
+     + '<li data-id="0" class="item"><i class="fas fa-trash-alt remove"></i></li>'
+     + '<li data-id="1" class="item"><i class="fas fa-trash-alt remove"></i></li>'
+     + '<li data-id="2" class="item"><i class="fas fa-trash-alt remove"></i></li>'
+     + '</ul>'
+     + '</div>';
+
+  const listTask = [
+    {
+      description: 'chocolate',
+      status: false,
+      index: 0,
+    },
+    {
+      description: 'candy',
+      status: true,
+      index: 1,
+    },
+    {
+      description: 'pie',
+      status: false,
+      index: 2,
+    },
+  ];
+  const localStorage = new LocalStorage();
+  localStorage.setDataLocalStorage(listTask);
+
+  const bundleRefreshHandlersAndUpdate = (list) => {
+    localStorage.setDataLocalStorage(list);
+    const firstArray = [];
+    const secondArray = [];
+    for (let i = 0; i <= list.length; i += 1) {
+      firstArray.push(i);
+    }
+    list.forEach((element) => secondArray.push(element.index));
+
+    const difference = firstArray.filter((x) => secondArray.indexOf(x) === -1);
+    const templateString = `li[data-id="${difference[0]}"]`;
+    const element = document.querySelector(templateString);
+    element.remove();
+  };
+
+  const checkNumberTask = () => [...document.querySelector('.container-list').children].length;
+
+  const element = document.querySelector('li[data-id="0"] .remove');
+
+  //  Act --------------------------------------------------------------------------->
+  const removeTask = (element) => {
+    const isRemoveIcon = element.classList.contains('remove');
+    if (isRemoveIcon) {
+      const listTasks = localStorage.getDataLocalStorage();
+      const id = parseInt(element.parentElement.dataset.id, 10);
+      listTasks.splice(id, 1);
+      bundleRefreshHandlersAndUpdate(listTasks);
+      return listTasks;
+    }
+    return false;
+  };
+
```

Comment:
- As I suggested you do for the add function, please I'd suggest you test the function responsible for removing items for your task lists instead of re-writing another function for this purpose. Please look into fixing this 🙏